projects/npm-tools/packages/npm-scripts/src/typescript/README.md

```diff
@@ -0,0 +1,62 @@
+# TypeScript build support in Liferay DXP
+
+If you're reading this page you may have arrived from a link inside a `tsconfig.json` file in [`liferay-portal`](https://github.com/liferay/liferay-portal). The purpose of this document is to describe how those configuration files are managed and how the TypeScript build process works in general.
+
+## High-level overview
+
+We use [TypeScript's](https://www.typescriptlang.org/) [project references feature](https://www.typescriptlang.org/docs/handbook/project-references.html) because it allows us to share type information between projects (corresponding to OSGi modules) without having to perform a global build of all TypeScript in DXP for every change.
+
+Each project writes out a set of `.d.ts` files and a `tsconfig.tsbuildinfo` file in a `types` directory that sits alongside `src` and should be committed to the repo. Any project which wishes to depend on another can use the previously committed type information, thus relying on the other project without needing to build it first. This will enable us to preserve fast "local" development inside each project even as the overall amount of TypeScript across the repo grows.
```

Comment:
About these docs: once gets merged I'll have some concrete config to link to and this can become a bit less abstract.

Reminder to self: clarify what happens with `tsconfig.tsbuildinfo` files (they can't be committed, so they have to go in `tmp/`, as explained in "refactor(npm-scripts): store tsbuildinfo in tmp/").